File: 102. Binary Tree Level Order Traversal/levelOrder.cpp
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
    
        if (root == nullptr) return {};
        
        vector<vector<int>> res = {{root->val}} ;
        vector<TreeNode*> cur_level = {root};

        while (!cur_level.empty()) {
            vector<int> vals;
            vector<TreeNode*> next_level;
            for (auto node: cur_level) {
                if (node->left) {
                    next_level.push_back(node->left);
                    vals.push_back(node->left->val);
                }
                if (node->right) {
                    next_level.push_back(node->right);
                    vals.push_back(node->right->val);
                }
            }

            if (!next_level.empty()) {
                cur_level.assign(next_level.begin(), next_level.end());
                res.push_back(vals);
            } else {
                break;
            }
        }
        return res;
    }
};

File: 102. Binary Tree Level Order Traversal/levelOrder.py
class Solution:

    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return

        res, cur_level = [[root.val]], [root]

        while cur_level:
            next_level, vals = [], []
            for node in cur_level:
                if node.left:
                    next_level.append(node.left)
                    vals.append(node.left.val)
                if node.right:
                    next_level.append(node.right)
                    vals.append(node.right.val)
            if next_level:
                cur_level = next_level.copy()
                res.append(vals)
            else:
                break
        return res